### DISCUSSION TITLE
CLI should have an option for quiet downloads

### DISCUSSION THREAD

**Discussion by danudey** [encode/httpx#2996] [Ideas]
httpx is a pretty great tool, but one thing I've noticed is that it's (apparently) impossible to use the CLI's download mode to get a progress bar but not show the HTTP response information. An example for when this would be useful is to use in a shell script.

Currently the 'best' way to do this that I know of is to use wget:

```wget --quiet --show-progress --progress=bar```

This works fine, but in cases where I have `httpx` (with CLI) installed already (or I'm already installing Python packages into the environment), the `httpx` progress bar is much nicer.

I can also envision cases where I want to fetch and display (or pipe) the contents of a file to stdout, but again without the extra headers.

What I propose is a CLI flag, `--show-headers`/`--hide-headers`, in which:

1. `--show-headers` is the default behavior
2. `--hide-headers` prevents the output of response headers in all cases (standard or download mode)
3. Adding `---show-headers` after `--hide-headers` resets the headers display; this is common behavior in other CLI utilities, and is default behavior provided by `click`.

I have added this functionality in my own fork and added unit tests for the functionality, for both standard and download mode. You can see the commit here:

https://github.com/danudey/httpx/commit/f1069d93302193a4d3094554b9064d69b75f7777

I haven't created a PR for this, as the 'Contributing' doc implies that a discussion should happen first, but I'm happy to create the PR if people feel this functionality could be useful.

**Comment by lovelydinosaur** [comment]
@danudey - This seems valuable to me yep. Thanks for the suggestion.

My inclination here would be supporting a `--quiet` option, mirroring the existing `--verbose` option.

(If we want to add `--show-headers`/`--hide-headers` then I'd suggest we'd probably also want to be looking comprehensively at all the output that `--verbose` provides, which seems like a more complex place to start from?)
